WooCommerce/WooCommerce.xcodeproj/project.pbxproj

```diff
@@ -5994,6 +5995,25 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		095040D72655531C001D08FA /* Check for nested frameworks */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+			);
+			name = "Check for nested frameworks";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "# Nested frameworks (i.e. having a Frameworks/ folder inside *.app/Frameworks/.framework) is invalid and will make the build be rejected during TestFlight validation. But this can happen especially due to an Xcode 12.4 UI bug when linking binary frameworks to the project which always embed the binary (bug that is fixed in Xcode 12.5)\n\nNESTED_FMKS=$(find \"${TARGET_BUILD_DIR}\"/WooCommerce.app/Frameworks/*.framework/Frameworks -name '*.framework') \nif [ -z \"$NESTED_FMKS\" ]; then\n    echo \"✅ No nested framework found, you're good to go!\"\nelse\n    for f in $NESTED_FMKS; do\n        nested=$(basename $f)\n        parent=$(basename $(dirname $(dirname $f)) .framework)\n        echo \"error: Found nested framework in $f -- Such a configuration is invalid and will be rejected by TestFlight. Please fix by choosing 'Do Not Embed' for the nested framework \\`$nested\\` within the \\`$parent\\` Xcode project containing it. You might need to use Xcode 12.5 to fix this, due to an Xcode 12.4 bug – see paNNhX-ee-p2\"\n    done\n    exit 1\nfi \n";
```

Comment:
@koke Seems like using `$TARGET_BUILD_DIR` was the right decision, according to [my favorite Build Settings reference](https://pewpewthespells.com/blog/buildsettings.html#target_build_dir) 👍 :
> Run Script build phases that operate on product files of the target that defines them should use the value of this build setting.

Reply:
Following up on the comment above, having the run script in an actual script also makes it simpler to read and edit it, as well as having a sensible diff to look at as opposed to a _single very long line_ (`:xcode:` :trollface: )

Reply:
Yeah I agree that I could have extracted that script into a file in Scripts/….sh, to make it more readable and diff-able 👍
That's actually something I usually do so not sure why I didn't that time around 😺